clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/WgsUbamTests.scala

```diff
@@ -85,21 +86,74 @@ trait WgsUbamTests { self: BaseIntegrationSpec =>
 
       it should s"handle upserts and queries for wgs-ubam location $location" in {
         for {
-          _ <- clioWebClient
+          upsertResponse <- clioWebClient
             .addWgsUbam(bearerToken, upsertKey, upsertData)
-            .map(_.status should be(StatusCodes.OK))
+          returnedClioId <- Unmarshal(upsertResponse).to[String]
           queryResponse <- clioWebClient
             .queryWgsUbam(bearerToken, queryData)
           outputs <- Unmarshal(queryResponse)
             .to[Seq[TransferWgsUbamV1QueryOutput]]
         } yield {
+          upsertResponse.status should be(StatusCodes.OK)
           outputs should have length 1
-          outputs.head should be(expected)
+          outputs.head should be(
+            expected.copy(clioId = Some(UUIDUtil.uuid(returnedClioId)))
+          )
         }
       }
     }
   }
 
+  it should "assign different clioIds to different WgsUbam upserts" in {
+    val upsertKey = TransferWgsUbamV1Key(
+      flowcellBarcode = "testClioIdBarcode",
+      lane = 2,
+      libraryName = s"library$randomId",
+      location = Location.GCP,
+    )
+    val upsertData = TransferWgsUbamV1Metadata(project = Some("testProject1"))
+
+    for {
+      clioId1 <- clioWebClient
+        .addWgsUbam(bearerToken, upsertKey, upsertData)
+        .flatMap(Unmarshal(_).to[String])
+      clioId2 <- clioWebClient
+        .addWgsUbam(
+          bearerToken,
+          upsertKey,
+          upsertData.copy(project = Some("testProject2"))
+        )
+        .flatMap(Unmarshal(_).to[String])
+    } yield {
+      clioId1 should not equal clioId2
+    }
+  }
+
+  it should "decode clioIds and supply a new one on upserts" in {
```

Comment:
The a new ClioId is assigned anyways, so it doesn't affect much. It just means that if someone takes the QueryOutput, changes a few fields, and upserts again, it won't fail because it cant parse the clioId